File: Mute/Audio/AppleScriptController.m
//
//  AppleScriptController.m
//  Mute
//
//  Created by Zachary Simone on 16/2/20.
//  Copyright © 2020 Zachary Simone. All rights reserved.
//

#import "AppleScriptController.h"

@implementation AppleScriptController

-(void) runCommand:(NSString*)command {
    NSAppleScript *script = [[NSAppleScript alloc] initWithSource:command];
    [script executeAndReturnError:nil]; // Not handling errors for now
}

-(void) setVolume:(NSInteger*)volume {
    NSString *volumeString = [NSString stringWithFormat:@"%ld", (long)*volume];
    [self runCommand:[[NSString alloc] initWithFormat:@"set volume input volume %@", volumeString]];
}

-(NSAppleEventDescriptor*) getResponse:(NSString*)command {
    NSDictionary *error = nil;
    NSAppleScript *script = [[NSAppleScript alloc] initWithSource:command];
    NSAppleEventDescriptor *result = [script executeAndReturnError:&error];
    NSLog(@"AppleScript command result: %@", result);
    return result;
}

@end

File: Mute/AppleScriptController.swift
//
//  RunAppleScriptCommand.swift
//  Mute
//
//  Created by Zachary Simone on 29/6/18.
//  Copyright © 2018 Zachary Simone. All rights reserved.
//

import Foundation

final class AppleScriptController {
    
    // MARK: Public
    
    static func setVolume(to volume: Int) {
        run(command: "set volume input volume \(volume)")
    }
    
    static func getResponse(command: String, completion: ((String)->())) {
        var error: NSDictionary?
        let script = NSAppleScript(source: command)
        if let output: NSAppleEventDescriptor = script?.executeAndReturnError(&error) {
            guard let response = output.stringValue else {
                completion("Error")
                return
            }
            completion(response)
        }
    }
    
    // MARK: Private helper
    
    private static func run(command: String) {
        let script = NSAppleScript(source: command)
        script?.executeAndReturnError(nil)
    }
}
